pkg/ipam/cell/cell.go

```diff
@@ -58,16 +82,38 @@ type ipamParams struct {
 	Sysctl              sysctl.Sysctl
 	EndpointManager     endpointmanager.EndpointManager
 	IPMasqAgent         *ipmasq.IPMasqAgent
+
+	DB         *statedb.DB
+	PodIPPools statedb.Table[podippool.LocalPodIPPool]
 }
 
-func newIPAddressManager(params ipamParams) *ipam.IPAM {
-	ipam := ipam.NewIPAM(params.Logger, params.NodeAddressing, params.AgentConfig, params.NodeDiscovery, params.LocalNodeStore, params.K8sEventReporter, params.NodeResource, params.MTU, params.Clientset, params.IPAMMetadataManager, params.Sysctl, params.IPMasqAgent)
+func newIPAddressManager(params ipamParams, c ipamConfig) (*ipam.IPAM, error) {
+	if c.OnlyMasqueradeDefaultPool && !params.AgentConfig.EnableBPFMasquerade {
+		return nil, fmt.Errorf("--only-masquerade-default-pool requires --enable-bpf-masquerade to be enabled")
+	}
```

Comment:
nit: This if statement could be moved into a separate "validateConfig" function that is invoked via `cell.Invoke` - but it's fine here as is.